### PR TITLE
Support exporting point clouds

### PR DESCRIPTION
**Pull request by tancik**
Pointclouds can now be exported from a trained nerf.

`ns-export pointcloud --load-config CONFIG --output-dir DIR`

### CODE REVIEW

File: nerfstudio/model_components/exporters.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+from nerfstudio.configs.base_config import Config  # pylint: disable=unused-import
+from nerfstudio.pipelines.base_pipeline import Pipeline
+
+CONSOLE = Console(width=120)
+
+
+def generate_point_cloud(
+    pipeline: Pipeline,
+    num_points: int = 1000000,
```
Comment (ethanweber) [inline nerfstudio/model_components/exporters.py:43]: 1e6

File: nerfstudio/model_components/exporters.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    remove_outliers: bool = True,
+    estimate_normals: bool = False,
+    rgb_output_name: str = "rgb",
+    depth_output_name: str = "depth",
+    use_bounding_box: bool = True,
+    bounding_box_min: Tuple[float, float, float] = (-1.0, -1.0, -1.0),
+    bounding_box_max: Tuple[float, float, float] = (1.0, 1.0, 1.0),
+) -> np.ndarray:
```
Comment (ethanweber) [inline nerfstudio/model_components/exporters.py:51]: o3d.geometry.PointCloud

File: pyproject.toml [hunk 3]
```diff
@@ -83,6 +84,7 @@ ns-download-data = "scripts.downloads.download_data:entrypoint"
 ns-train = "scripts.train:entrypoint"
 ns-eval = "scripts.eval:entrypoint"
 ns-render = "scripts.render:entrypoint"
+ns-export = "scripts.exporter:entrypoint"
```
Comment (ethanweber) [inline pyproject.toml:87]: scripts.export:entrypoint

File: scripts/exporter.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    """Path to the config YAML file."""
+    output_dir: Path
+    """Path to the output directory."""
+
+
+@dataclass
+class ExportPointCloud(Exporter):
+    """Class to export meshes with the poisson algorithm."""
+
+    num_points: int = 1000000
```
Comment (ethanweber) [inline scripts/exporter.py:36]: 1e6

**Review by ethanweber — APPROVED** [review]
LGTM. Just a few comments.

File: pyproject.toml (hunk 3 above)
Comment (ethanweber) [reply pyproject.toml:87]: Oh nevermind.

File: nerfstudio/model_components/exporters.py (hunk 1 above)
Comment (tancik) [reply nerfstudio/model_components/exporters.py:43]: 1e6 is a float, not an int. I can add the typecast, but at that point it seems a bit excessive.

**Pull request merged**